Resources/WebsiteTextServiceImpl.cs

```diff
@@ -82,6 +88,18 @@ public override void CreateWebsiteTexts(IWebsiteTextSource textSource)
             }
         }
 
+        private static void AddOrUpdateValue(IWebsiteTextSource textSource, Website website, KeyValuePair<string, string> textValue, string textKey)
+        {
+            if (textSource.UpdateExistingTexts)
+            {
+                website.Texts.AddOrUpdateValue(textKey, textValue.Key, textValue.Value);
+            }
+            else if (!website.Texts.Keys.Contains(textKey))
+            {
+                website.Texts.AddOrUpdateValue(textKey, textValue.Key, textValue.Value);
+            }
```

Comment:
To make this more readable you could create a parameter:
`var textIsNew = !website.Texts.Keys.Contains(textKey);`

Then you can make a single clear if-statement with `textIsNew || textSource.UpdateExistingTexts`

Reply:
Quick fix done